dotnet/src/SemanticKernel.Abstractions/Filters/Prompt/PromptRenderContext.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Class with data related to prompt rendering.
+/// </summary>
+[Experimental("SKEXP0001")]
+public sealed class PromptRenderContext
+{
+    private string? _renderedPrompt;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PromptRenderContext"/> class.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Microsoft.SemanticKernel.Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="function">The <see cref="KernelFunction"/> with which this filter is associated.</param>
+    /// <param name="arguments">The arguments associated with the operation.</param>
+    internal PromptRenderContext(Kernel kernel, KernelFunction function, KernelArguments arguments)
+    {
+        Verify.NotNull(kernel);
+        Verify.NotNull(function);
+        Verify.NotNull(arguments);
+
+        this.Kernel = kernel;
+        this.Function = function;
+        this.Arguments = arguments;
+    }
```

Comment:
We can, but I would say to keep context classes as lightweight as possible, and we can always extend it later if needed. Do you have some scenarios when these properties will be useful?

Reply:
The use case I thinking about is during prompt rendering we don't have access to those values so this would allow you to write a filter that changed the rendered prompt.
But agreed let's leave it as is for now and I'll think the scenario through a bit more.